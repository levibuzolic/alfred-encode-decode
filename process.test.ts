import Process from './process';

jest.mock('./qjs', () => ({
  cwd: () => '/mock/cwd',
}));

describe('decode', () => {
  ['SGVsbG8=', 'x'].forEach((input) => {
    test(`input: ${input}`, () => {
      const result = new Process('decode', input);
      expect(result.scriptFilters).toMatchSnapshot();
    });
  });
});

describe('encode', () => {
  ['Hello'].forEach((input) => {
    test(`input: ${input}`, () => {
      const result = new Process('encode', input);
      expect(result.scriptFilters).toMatchSnapshot();
    });
  });
});
