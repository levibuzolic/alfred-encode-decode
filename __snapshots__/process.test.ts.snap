// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decode input: SGVsbG8= 1`] = `
Object {
  "items": Array [
    Object {
      "arg": "Hello",
      "icon": undefined,
      "subtitle": "Base64",
      "title": "Hello",
      "valid": true,
    },
    Object {
      "arg": "Hello",
      "icon": undefined,
      "subtitle": "Base64 URL Safe",
      "title": "Hello",
      "valid": true,
    },
    Object {
      "arg": "SGVsbG8=",
      "icon": undefined,
      "subtitle": "HTML Entities",
      "title": "SGVsbG8=",
      "valid": true,
    },
  ],
}
`;

exports[`decode input: x 1`] = `
Object {
  "items": Array [
    Object {
      "arg": "x",
      "icon": undefined,
      "subtitle": "HTML Entities",
      "title": "x",
      "valid": true,
    },
    Object {
      "arg": "",
      "icon": Object {
        "path": "/mock/cwd/invalid.png",
      },
      "subtitle": "Base64",
      "title": "No Result",
      "valid": false,
    },
    Object {
      "arg": "",
      "icon": Object {
        "path": "/mock/cwd/invalid.png",
      },
      "subtitle": "Base64 URL Safe",
      "title": "No Result",
      "valid": false,
    },
  ],
}
`;

exports[`encode input: Hello 1`] = `
Object {
  "items": Array [
    Object {
      "arg": "SGVsbG8=",
      "icon": undefined,
      "subtitle": "Base64",
      "title": "SGVsbG8=",
      "valid": true,
    },
    Object {
      "arg": "SGVsbG8",
      "icon": undefined,
      "subtitle": "Base64 URL Safe",
      "title": "SGVsbG8",
      "valid": true,
    },
    Object {
      "arg": "Hello",
      "icon": undefined,
      "subtitle": "HTML Entities",
      "title": "Hello",
      "valid": true,
    },
  ],
}
`;
